test/framework/src/main/java/org/elasticsearch/test/transport/CapturingTransport.java

```diff
@@ -97,7 +97,7 @@ public TransportService createCapturingTransportService(Settings settings, Threa
                                                             @Nullable ClusterSettings clusterSettings, Set<String> taskHeaders) {
         StubbableConnectionManager connectionManager = new StubbableConnectionManager(new ConnectionManager(settings, this, threadPool),
             settings, this, threadPool);
-        connectionManager.setDefaultNodeConnectedBehavior((cm, discoveryNode) -> nodeConnected(discoveryNode));
```

Comment:
This is a port of some changes from the `zen2` branch for easier future merging. Please do not back this out.